app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user&.authenticate(params[:session][:password])
```

Comment:
Question 1: `&` có nghĩa là gì e nhỉ
Question 2: Hàm `authenticate` hoạt động thế nào

Reply:
- Với & thì nó là 1 toán tử giúp khi mình gọi phương thức trên 1 đối chỉ khi đối tượng đó không phải nil, nếu nó nil thì giúp biểu thức trả về nil thay vì ném ra lỗi; như trong trường hợp trên nó sẽ gọn hơn mà tương tự so với viết như thế này: user && user.authenticate(params[:session][:password]). 

- Còn hàm authenticate để xác thực mật khẩu của người dùng: đầu tiên nó sẽ nhận mật khẩu nhập từ người dùng, sau đó mã hóa mk đó rồi so sánh với mật khẩu đã được mã hóa trong db, nếu khớp thì  sẽ trả về người dùng còn không sẽ trả về false